### ISSUE TITLE
Google Analytics to allow for websites that conduct transactions in multiple currencies

### ISSUE THREAD

**Issue by XigenDominic** [magento/magento2#6709]
Google Analytics to allow for websites that conduct transactions in multiple currencies.  The ecommerce plugin allows you to specify the local currency of the transaction.
### Preconditions
1. Magento v2.1.1
### Steps to reproduce
1. Enable Google Analytics
2. Complete transaction
### Expected result
1. Option to enable specifying local currency
2. Local currency included in tag - https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#multicurrency
### Actual result
1. Local currency not included within tag

**Comment by TomashKhamlai** [comment]
Let me improve your issue. I will try to guess what actions you performed. Correct me if I am wrong.
# Preconditions

1. Magento v2.1.1 is installed.
2. A simple product is created.
# Steps to reproduce

1. Go to Stores -> Configuration -> Sales -> Google API.
2. Change _Enable_ from 'No' to 'Yes'.
3. Visit [this page](https://www.google.com/analytics/web/#home/) to obtain your account number. It should look like UA-XXXXXXXXX-X. Use [this guide](https://support.google.com/analytics/answer/1042508) for reference.
4. Change the value of **Account Number**. Copy and paste it from the previous step.
5. Go to Storefront as a guest.
6. Add simple product to a cart.
7. Proceed to checkout.
8. **Please provide URL of the page where you are expecting the result or add some additional instructions.**
# Expected result:
 **We love screenshots!**

> Option to enable specifying local currency
> Local currency included in tag - https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#multicurrency

Actual result

> Local currency not included within tag

**Comment by XigenDominic** [comment]
That's it.  

8. Taken to checkout success page and in GA markup you should get 

ga('set', 'currencyCode', 'EUR'); // Set tracker currency to Euros for example if order was in euros

**Comment by TomashKhamlai** [comment]
Thank you for your submission. Unfortunately, I can not submit this issue as a bug in our internal bug tracking system. But don't give up! You have two possible options to do next.

The first one is to visit our forum where people ask for feature improvements. We recently made some changes to the way we process GitHub submissions to more quickly identify and respond to core code issues. Feature Requests and Improvements should now be submitted to the new  [Magento 2 Feature Requests and Improvements forum](https://community.magento.com/t5/Magento-2-Feature-Requests-and/idb-p/feature-requests) (see details [here](https://community.magento.com/t5/News-Announcements/Improvements-to-GitHub-Management/m-p/44572#M96)).

The second one, you can try to fix it. I will give you an idea. I just have to warn you that **fixing issues is not my area of responsibility and my code snippet may be buggy**, however, it worked for me on Magento v2.1.2 that I installed using composer.

How to fix this issue:

1. Create a backup of your store.
2. Download the code from your server using popular FTP/SFTP clients.
3. Open in your favorite text editor ./vendor/magento/module-google-analytics/Block/Ga.php
4. Find this line of code `$result[] = "ga('require', 'ec', 'ec.js');";` it should be near line ~108.
5. Paste after this line the next snippet:
```
        foreach ($collection as $order) {
            $currencyCode = $order->getBaseCurrencyCode();
        }

        $result[] = sprintf(
            "ga('set', 'currencyCode', '" . $currencyCode . "')"
        );

```
6. Save it and upload back to your server.

I got this in my browser as a result:
![result](https://user-images.githubusercontent.com/11827230/27914009-8c9b7c46-626a-11e7-92a3-3d9a9ef2c2bb.png)

**Comment by XigenDominic** [comment]
That's great.  Although as your editing vendor could this not get overwritten by a composer update?

**Comment by TomashKhamlai** [comment]
@XigenDominic yes composer update will overwrite this. There are people from Magento that will review my answers and correct me if I was wrong in something. We requested this check, so please do not close this issue and wait for their response.

I voted up [your idea](https://community.magento.com/t5/Magento-2-Feature-Requests-and/Currency-code-set-on-google-analytics-block-script/idi-p/70399) on the forum. Thank you for your patience!

**Comment by korostii** [comment]
Hi @TomashKhamlai, I really appreciate the amount of compassion and effort you put into your job, personally. Sadly, this seems to be a rare quality among the other guys doing the issue sorting here.

The point of having this improvement added into the core is exactly in order to avoid having it yourself either to vendor (after each update) or via some kind of module-level or theme-level overrides etc.

BTW, Why does it have a `needs update` label? shouldn't it be marked as `moved to forums` instead?

**Comment by korostii** [comment]
_The following is grumpy complaint at the current issue management policies in general. Please, please don't take it personally_ :) 
_As a matter of fact, feel free to ignore it completely, just like the others would._ 

> Unfortunately, I can not submit this issue as a bug in our internal bug tracking system.

Your suggested bugfix is a clear improvement, it's fairly easy to implement, I don't think there can be any sane argument against it other than "just put it at the back of the queue", and yet "you cannot submit it". 

Therefore, you let it rot on this new circle of hell called "Feature Requests and Improvements forum" for a few more months, as if _that_'s gonna do any good. Is there at least a single suggested improvement on that forum that actually got implemented? I only see "new" and "investigating" (whatever that means). 

I really don't see any benefit from having it on forums (rather than here, with labels `improvement` and `up for grabs`) if it's effectively ignored by Magento, Inc. anyway. It, sort of, kills the enthusiasm, when all the issue reports and suggestions get dismissed and overlooked for months straight. It's just sad to see.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-71539

**Comment by DominicWatts** [comment]
👍

**Comment by magento-engcom-team** [comment]
@XigenDominic, thank you for your report.
The issue is already fixed in develop branch
But we will consider to backport the fix to patch releases

**Comment by korostii** [comment]
Hi @magento-engcom-team

Considering that `develop` is now corresponding to 2.3.0 version which is at least one year from release, how closing this issue makes any sense right now?

Could you please publicly share your current guidelines regarding closing acknowledged\reproduced issues and github issue flow in general? 
That would be most helpful in order to understand what's going on here.

Thanks in advance.

**Comment by magento-engcom-team** [comment]
Hi @korostii 
Please note that this issue report remains open for now. It may be closed when fix is backported into `2.2.x` or `2.3` is released. 
We are considering to publicly announce guidelines for the GitHub issues tracking in future.
Thank you for collaboration.

**Comment by magento-engcom-team** [comment]
@XigenDominic, thank you for your report.
The issue is already fixed in develop branch
But we will consider to backport the fix to patch releases

**Comment by XigenDominic** [comment]
I created the pull request that was merged into develop. I did it on my personal account.

**Comment by korostii** [comment]
@XigenDominic 
FYI if you'd like to expedite this bugfix further and have spare time for one more PR, it's encouraged to submit an extra "backport" pull request to `2.1-develop` (and\or `2.2-develop`, probably) with very same changes as well (now that your PR to `develop` is reviewed, accepted and merged).

Don't have a link handy, but this policy was publicly announced on the Magento Forums a few months back.

**Comment by magento-engcom-team** [comment]
Hi @XigenDominic the issue has been fixed in 2.2-develop branch and will be available with release 2.2.1 soon
